src/main/java/io/vertx/core/eventbus/impl/clustered/Serializer.java

```diff
@@ -127,30 +124,26 @@ void close() {
         tasks.remove().internalPromise.tryFail("Context is closing");
       }
     }
+  }
 
-    private class SerializedTask<U> implements Handler<AsyncResult<U>> {
+  private class SerializedTask<U> {
```

Comment:
why are you removing this ? it allows to save a instance creation

Reply:
Because with the changes in this PR, it's not needed anymore.

Reply:
I mean, it's not needed to save an instance creation anymore.

Reply:
`task.future().onComplete(v -> processed())` does instantiate a lambda unless we cache it.

If `SerializerTask` is an handler then it could call in handle the `processed()` method

Reply:
I've updated the PR